### PR TITLE
Include accessible private members from outer types in Intellisense

### PR DESCRIPTION
**Pull request by cston**
Restores original fix for #2639 with the additional change to avoid re-calculating base types.

### CODE REVIEW

**Comment by cston** [comment]
@Pilchie, @rchande

**Comment by rchande** [comment]
LGTM, modulo tests for VB. Might be good to submit a build to rosperf-ssv to get perf numbers (I can show you how, if you haven't done so before).

**Comment by cston** [comment]
@pharring

**Comment by pharring** [comment]
If you push to master, we'll get a read on perf in the rolling IDE runs.

File: src/Workspaces/Core/Portable/Recommendations/AbstractRecommendationService.cs [hunk 1]
```diff
@@ -21,77 +21,117 @@ internal abstract class AbstractRecommendationService : IRecommendationService
...
-            var namespaceSymbol = symbol as INamespaceSymbol;
-            if (namespaceSymbol != null)
+            private IEnumerable<INamedTypeSymbol> GetEnclosingTypeBases()
             {
-                return namespaceSymbol.ContainsAccessibleTypesOrNamespaces(context.SemanticModel.Compilation.Assembly);
+                if (_lazyEnclosingTypeBases == null)
+                {
+                    var enclosingType = _context.SemanticModel.GetEnclosingNamedType(_context.LeftToken.SpanStart, _cancellationToken);
+                    _lazyEnclosingTypeBases = (enclosingType == null) ?
+                        Enumerable.Empty<INamedTypeSymbol>() :
```
Comment (pharring) [inline src/Workspaces/Core/Portable/Recommendations/AbstractRecommendationService.cs]: Prefer
```C#
SpecializedColletions.EmptyEnumerable<INamedTypeSymbol>()

**Comment by pharring** [comment]
:+1:

**Comment by jaredpar** [comment]
:+1: 

Finally!  I can edit my VsVim unit tests again with full Intellisense :)

**Comment by cston** [comment]
@dotnet-bot Retest this, please

**Comment by Pilchie** [comment]
:+1:

**Pull request merged**